contractcourt/chain_watcher.go

```diff
@@ -373,6 +427,28 @@ func (c *chainWatcher) closeObserver(spendNtfn *chainntnfs.SpendEvent) {
 			return
 		}
 
+		// Fetch the current known commit height for the remote party,
+		// and their pending commitment chain tip if it exist.
+		remoteStateNum := remoteCommit.CommitHeight
+		remoteChainTip, err := c.cfg.chanState.RemoteCommitChainTip()
+		if err != nil && err != channeldb.ErrNoPendingCommit {
+			log.Errorf("unable to obtain chain tip for "+
+				"ChannelPoint(%v): %v",
+				c.cfg.chanState.FundingOutpoint, err)
+			return
+		}
+
+		// Now that we have all the possible valid commitments, we'll
+		// make the CommitSet the ChannelArbitrator will need it in
+		// order to carry out its duty.
+		commitSet := CommitSet{
+			LocalCommit:  *localCommit,
+			RemoteCommit: *remoteCommit,
+		}
+		if remoteChainTip != nil {
+			commitSet.RemotePendingCommit = &remoteChainTip.Commitment
+		}
```

Comment:
Shouldn't c.cfg.chanState.LatestCommitments() return the remotePendingCommit too, instead of fetching it separately? Or expose it as a field on OpenChannel?

Reply:
Possibly, they could very well be combined but I don't think we need to do that in this PR (this is just code that was moved around). It isn't a field since this needs to be read from disk and my not be present at all.